packages/libjson-c/libjson-c.spec

```diff
@@ -0,0 +1,53 @@
+%global jsonc_ver 0.18
+%global jsonc_rel 20240915
+
+Name: %{_cross_os}libjson-c
+Version: %{jsonc_ver}
+Release: %{jsonc_rel}%{?dist}
+Summary: Library for JSON
+License: MIT
+Source0: https://github.com/json-c/json-c/archive/refs/tags/json-c-%{jsonc_ver}-%{jsonc_rel}.tar.gz
+BuildRequires: %{_cross_os}glibc-devel
+
+%description
+%{summary}.
+
+%package devel
+Summary: Files for development using the library for JSON
+%description devel
+%{summary}.
+
+%prep
+%autosetup -n json-c-json-c-%{jsonc_ver}-%{jsonc_rel} -p1
+
+%build
+%{cross_cmake} . \
+  -DBUILD_APPS:BOOL=OFF \
+  -DBUILD_SHARED_LIBS:BOOL=ON \
+  -DBUILD_STATIC_LIBS:BOOL=OFF \
+  -DDISABLE_BSYMBOLIC:BOOL=OFF \
+  -DDISABLE_EXTRA_LIBS:BOOL=ON \
+  -DDISABLE_JSON_POINTER:BOOL=ON \
+  -DDISABLE_THREAD_LOCAL_STORAGE:BOOL=OFF \
+  -DDISABLE_WERROR:BOOL=OFF \
+  -DENABLE_RDRAND:BOOL=ON \
+  -DENABLE_THREADING:BOOL=ON \
+  -DCMAKE_INSTALL_PREFIX:PATH=%{_cross_prefix} \
+  -G Ninja
+
+cmake --build .
```

Comment:
Are these two steps (`_cross_cmake` and `cmake build`) required in separate lines? If `_cross_cmake` is used to generate the actual `Make` files, why not move that to `%prep`?

Reply:
> Are these two steps (_cross_cmake and cmake build) required in separate lines?

Even if they're not, I prefer them to be separate for symmetry with autotools (`configure` then `make`), meson (`meson setup` then `meson build`), etc.

> If _cross_cmake is used to generate the actual Make files, why not move that to %prep?

`%prep` is only meant to prepare the source tree for building. It's not meant to encompass build steps, and by convention the entry point to the build is that first `configure` / `meson setup` / `cmake` invocation. In the `configure` case this often involves building and running a lot of test programs; it isn't a "pure" operation on files in the way that `%prep` ideally is.

With sufficient motivation you could run build logic in the `%prep` stage or patch sources in the `%build` stage (and our `grub` and `glibc` packages kind of do that, respectively). In extreme cases (like our `libgcc`) the entire build can happen elsewhere. But it's not the convention and it makes the spec "weird" when you do that.